rero_ils/modules/libraries/api.py

```diff
@@ -193,11 +193,12 @@ def _has_is_open(self):
                     return True
         if exception_dates := self.get('exception_dates'):
             for exception_date in exception_dates:
-                start_date = date_string_to_utc(exception_date['start_date'])
-                # avoid next_open infinite loop if an open exception date is
-                # in the past
-                if start_date > datetime.now(pytz.utc):
-                    return True
+                if exception_date['is_open']:
+                    date = date_string_to_utc(exception_date['start_date'])
+                    # avoid next_open infinite loop if an exception date
+                    # exists in the past
+                    if date > datetime.now(pytz.utc):
+                        return True
```

Comment:
maybe I will wrote 
```python
...
current_timestamp = datetime.now(pytz.utc)
for date in filter(lambda d: d['is_open'], self.get('exception_dates', []):
    start_date = date_string_to_utc(exception_date['start_date'])
    # avoid next_open infinite loop if an open exception date is
    # in the past
    if start_date > current_timestamp:
        return True
...
```

But your code should work too

Reply:
@zannkukai Thanks, you're right it's more elegant. :)